app/parse.py

```diff
@@ -8,8 +15,46 @@ class Quote:
     tags: list[str]
 
 
+def get_single_quotes(quote_soup: Tag) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=quote_soup.select_one(".author").text,
+        tags=[tag.text for tag in quote_soup.select(".tag")]
+    )
+
+
+def get_single_page_quotes(page_soup: BeautifulSoup) -> list[Quote]:
+    quotes = page_soup.select(".quote")
+    return [get_single_quotes(quote_soup) for quote_soup in quotes]
+
+
+def parse_quotes(url: str) -> list[Quote]:
```

Comment:
The code seems correct for navigating pages, but be cautious of a potential infinite loop if there is a malformed or unexpected response. Adding some error handling might be beneficial.